aws/data_source_aws_secretsmanager_secret_rotation_test.go

```diff
@@ -0,0 +1,110 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccDataSourceAwsSecretsManagerSecretRotation_Basic(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_secretsmanager_secret_rotation.test"
+	datasourceName := "data.aws_secretsmanager_secret_rotation.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t); testAccPreCheckAWSSecretsManager(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsSecretsManagerSecretRotationConfig_NonExistent,
+				ExpectError: regexp.MustCompile(`not found`),
+			},
+			{
+				Config: testAccDataSourceAwsSecretsManagerSecretRotationConfig_Default(rName, 7),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceAwsSecretsManagerSecretRotationCheck(datasourceName, resourceName),
```

Comment:
This custom function can be replaced with usage of standard Terraform Plugin SDK check functions:

```suggestion
					resource.TestCheckResourceAttrPair(datasourceName, "rotation_enabled", resourceName, "rotation_enabled"),
					resource.TestCheckResourceAttrPair(datasourceName, "rotation_lambda_arn", resourceName, "rotation_lambda_arn"),
					resource.TestCheckResourceAttrPair(datasourceName, "rotation_rules.#", resourceName, "rotation_rules.#"),
```